### ISSUE TITLE
Mutable Option can't be change to None once it was set to Some

### ISSUE THREAD

**Issue by Booksbaum** [fable-compiler/Fable#2147]
### Description
A mutable option can't be changed to `None` once it was set to `Some`

### Repro code

```fsharp
let mutable v: int option = None
printfn "None = %A" v
v <- Some 1
printfn "Some 1 = %A" v
v <- None
printfn "None = %A" v
```

[REPL](https://fable.io/repl/#?code=DYUwLgBAtgrmCGAjUEBuAuCBLAdpA9gA5hb44QC8EAcmSALABQhATrmAGbkBEtOIlCAFIAgtzRNUEADwBaCAGV8UAQEYmrdlwjclKiKsGjxqSTPl8GzNnm286RsWiA&html=Q&css=Q)  
(versions: `REPL: 2.1.1; Fable: 2.10.1`)

### Expected and actual results

Expected output:
> None = null  
> Some 1 = 1  
> None = null  

or

> None = undefined  
> Some 1 = 1  
> None = undefined  

Actual output:
> None = undefined  
> Some 1 = 1  
> None = 1

### Related information
Tested with fable-compiler `2.12.0`, `2.10.0` & `2.10.1` (REPL)

The bug was most likely introduced by changing `None` from `null` to `undefined` in 2b365f505db41d1d9d6f32ea975bd194b164584f.
fable-compiler version `2.9.0` still works, while `2.10.0` doesn't. `2.10.0` contains the change from `null` to `undefined`. When changing the option to None `2.10.0` emits `v(undefined)` while `2.9.0` emits `v(null)`.

**Comment by ncave** [comment]
Fixed in #2148.

**Comment by MangelMaxime** [comment]
@Booksbaum Should be fixed in version 2.13.0
